### DISCUSSION TITLE
Maps and Set in storage without serialize / deserialize (deprecated)

### DISCUSSION THREAD

**Discussion by wcastand** [pmndrs/zustand#1510] [Q&A]
Hi how do i persis tmy Map/Set now that persist has deprecated seriablize/deserialized

couldn't find the solution in the new persist doc for Map/Set. No mention of persist in the Map/Set doc either.

until the update i was doing this:
```		
serialize: (state) =>
			JSON.stringify({
				state: {
					...state.state,
					transactions: Array.from(state.state.transactions.entries()),
				},
			}),
		deserialize: (state) => ({
			state: {
				...JSON.parse(state).state,
				transactions: new Map(JSON.parse(state).state.transactions),
			},
		}),
```

**Comment by dai-shi** [comment]
Thanks for opening this up.
I think I overlooked the custom (de)serialize usage.

Here's something to work:
```js
  storage: {
    getItem: (name) => {
      const str = localStorage.getItem(name)
      return {
        state: {
          ...JSON.parse(str).state,
          transactions: new Map(JSON.parse(str).state.transactions),
        },
      }
    },
    setItem: (name, newValue) => {
      const str = JSON.stringify({
        state: {
          ...newValue.state,
          transactions: Array.from(newValue.state.transactions.entries()),
        },
      })
      localStorage.setItem(name, str)
    },
    removeItem: (name) => localStorage.removeItem(name),
  },
```

If that actually works for your use case, would you like to open a PR to improve docs please?

**Comment by wcastand** [comment]
i forked and was writting the PR but i repo create new file `.yarn` and `.yarnrc.yml`

My team use yarn 3 and i'm guessing you're not so maybe you need to add a config to specify the version of yarn that need to be use when we do a `yarn install` :)

**Reply by dai-shi** [reply]
In package.json? We use yarn 1 for all `zustand`, `jotai`, and `valitio` repos. Should fix all of them...

**Reply by wcastand** [reply]
yes in our project in the root package.json we have `"packageManager": "yarn@3.2.4"`
this you have nothing defined and my yarn is by default yarn 3, that's what yarn use as a default version. I don't know if that makes sense.

i just tried to do `yarn set version classic` to force yarn v1 but it still create `.yarn` and `.yarnrc.yml` so not sure how i can work on the repo without making a massive pr. i'll try to clean the PR since it's only doc in theory

the PR: https://github.com/pmndrs/zustand/pull/1512
